### PR TITLE
LGA 3318: Implement basic templates and routing

### PR DESCRIPTION
**Pull request by BenMillar-MOJ**
### Adds prototype page templates and routing for three user journeys
1. In scope for CLA, go to means test, including onward questions
Discrimination journey - select ‘Disability’
Select: Discrimination
Then select: Work
Then select: Disability, health condition, mental health condition
End point: https://staging.checklegalaid.service.gov.uk/legal-aid-available?hlpas=no

2. In scope for CLA, go direct to ‘Contact CLA’
Domestic abuse journey, including onward question
Select: Domestic abuse
Then select: Help to protect you or your children
Then select: Yes (question ‘Are you at immediate risk of harm?’ radios: Yes / No)
End point: https://staging.checklegalaid.service.gov.uk/contact?hlpas=no

3. In scope for FALA, no means test (why do we means test anyone for FALA???)
Select: Clinical negligence
Endpoint: https://staging.checklegalaid.service.gov.uk/scope/refer/legal-adviser?category=clinneg&hlpas=no

Screenshot 2024-10-18 at 11 51 11
Screenshot 2024-10-18 at 11 51 15
Screenshot 2024-10-18 at 11 51 24

## Any other changes that would benefit highlighting?

- Introduces a GOVUK Radio Widget that overrides the WTForms implementation to use the required label CSS Class.
- Now uses `app_before_request` for the maintenance pages, so they are shown regardless of the blueprint used.
- `get_gtm_anon_id` is now registered to the app so it applies to all blueprints
- Puts a temporary watermark over the pages to indicate this is just a prototype.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by TawneeOwl — APPROVED** [review]
Looks awesome! Thank you!

File: app/categories/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from .domestic_abuse import bp as domestic_abuse_bp
+from .discrimination import bp as discrimination_bp
+
+bp = Blueprint("categories", __name__, template_folder="templates")
+bp.register_blueprint(domestic_abuse_bp)
+bp.register_blueprint(discrimination_bp)
+
+from app.categories import routes  # noqa: E402,F401
+
+__all__ = ["bp"]
```
Comment (said-moj) [inline app/categories/__init__.py:11]: What is this doing?

Comment (BenMillar-MOJ) [reply app/categories/__init__.py:11]: It isn't really needed here, it exists to decide what to export when you use `from module import *`. 
I have removed it.

File: app/categories/discrimination/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from flask import Blueprint
+
+bp = Blueprint(
+    "discrimination",
+    __name__,
+    template_folder="./templates",
```
[the remark this answers is not in the document]
Comment (BenMillar-MOJ) [reply app/categories/discrimination/__init__.py:6]: My reasoning was that it makes sense to keep pages in the same area as the blueprint, allowing you to more easily find the relevant page and avoiding having a single directory with a large number of templates.

However, I am happy to discuss this and change course.

[the remark this answers is not in the document]
Comment (BenMillar-MOJ) [reply app/categories/discrimination/__init__.py:6]: Discussed during dev retro.

**Review by digitalali-moj — APPROVED** [review]
Thanks guys, this is amazing stuff. I understand we might have to iterate along the way, but a brilliant start! Reminder just to look at the tests.

**Pull request merged**